app/models/tag.rb

```diff
@@ -674,302 +654,99 @@ def recategorize(new_type)
-                find_in_batches do |batch|
-      IndexQueue.enqueue_ids(Pseud, batch.map(&:pseud_id), :background)
-    end
-  end
-
-  # reindex all works that are tagged with this tag or its subtags or synonyms (the filter_taggings table)
-  # if work_ids are passed in, those will be used (eg if we need to save the ids before making changes, then
-  # reindex after the changes are done)
-  def reindex_all_works(work_ids = [])
-    if work_ids.empty?
-      work_ids = all_filtered_work_ids
-    end
-    RedisSearchIndexQueue.queue_works(work_ids, priority: :low)
-  end
-
-  # In the case of works, the filter_taggings table already collects all the things tagged
-  # by this tag or its subtags/synonyms
-  def all_filtered_work_ids
-    # all synned and subtagged works should be under filter taggings
-    # add in the direct works for any noncanonical tags
-    (self.filter_taggings.where(filterable_type: "Work").pluck(:filterable_id) +
-      self.works.pluck(:id)).uniq
-  end
-
-  # Reindex all series (series_ids argument works as above)
-  def reindex_all_series(series_ids = [])
-    if series_ids.empty?
-      series_ids = all_filtered_series_ids
-    end
-    IndexQueue.enqueue_ids(Series, series_ids, :background)
-  end
-
-  # Series get their filters through works, so we go through SerialWork, which has
-  # both work and series ids
-  def all_filtered_series_ids
-    SerialWork.where(work_id: all_filtered_work_ids).pluck(:series_id).uniq
-  end
-
-  # In the case of external works, the filter_taggings table already collects all the
-  # things tagged by this tag or its subtags/synonyms
-  def all_filtered_external_work_ids
-    # all synned and subtagged external works should be under filter taggings
-    # add in the direct external works for any noncanonical tags
-    (filter_taggings.where(filterable_type: "ExternalWork").pluck(:filterable_id) +
-      external_works.pluck(:id)).uniq
-  end
-
-  # Reindex all external works (external_work_ids argument works as above)
-  def reindex_all_external_works(external_work_ids = [])
-    if external_work_ids.empty?
-      external_work_ids = all_filtered_external_work_ids
-    end
-    IndexQueue.enqueue_ids(ExternalWork, external_work_ids, :background)
-  end
-
-  # Reindex all bookmarks (bookmark_ids argument works as above)
-  def reindex_all_bookmarks(bookmark_ids = [])
-    if bookmark_ids.empty?
-      bookmark_ids = all_bookmark_ids
+  before_update :reindex_all_for_name_or_type_change
+  def reindex_all_for_name_or_type_change
+    return unless name_changed? || type_changed?
+
+    reindex_pseuds = (type == "Fandom") || (type_was == "Fandom")
+    async_after_commit(:reindex_all, reindex_pseuds)
+  end
+
+  # Reindex anything even remotely related to this tag. This is overkill in
+  # most cases, but necessary when something fundamental like the name or type
+  # of a tag has changed.
+  def reindex_all(reindex_pseuds = false)
```

Comment:
I find this method name confusing, because it's for reindexing everything related to this tags, not for reindexing all tags (like Searchable's `reindex_all`). Can we rename it and `reindex_all_for_name_or_type_change`?

`reindex_anything_related` but I feel like there's better noun.

Reply:
Just brainstorming here:
- `reindex_related_records`
- `reindex_related_documents`
- `reindex_related`
- `reindex_associated`
- `reindex_references`

Any of those sound good?

Reply:
`reindex_associated` sounds good, matches Rails' terminology too.

https://github.com/rails/rails/commit/4e076b03b6312b928a845f9a31288d10d7243338